ArchiSteamFarm/Bot.cs

```diff
@@ -309,6 +314,45 @@ internal sealed class Bot : IDisposable {
 			return await BotDatabase.MobileAuthenticator.HandleConfirmations(confirmations, accept).ConfigureAwait(false);
 		}
 
+		private async Task AcceptDigitalGiftCards() {
+			lock (GiftCardsSemaphore) {
+				if (ProcessingGiftsScheduled) {
+					return;
+				}
+
+				ProcessingGiftsScheduled = true;
+			}
+
+			await GiftCardsSemaphore.WaitAsync().ConfigureAwait(false);
+
+			try {
+				lock (GiftCardsSemaphore) {
+					ProcessingGiftsScheduled = false;
+				}
+
+				HashSet<ulong> gids = await ArchiWebHandler.GetDigitalGiftCards().ConfigureAwait(false);
+				if ((gids == null) || (gids.Count == 0)) {
+					return;
+				}
+
+				foreach (ulong gid in gids) {
```

Comment:
You're not making use of `HandledGifts` here, add `.Where(gid => !HandledGifts.Contains(gid))` to the `gids` so you skip handling the same one several times in a row.